### PR TITLE
Dashboard Settings Item [PRO-982]

### PR DESCRIPTION
**Pull request by parag-chauhan**
## Description
- A Settings Menu item that allows a user to launch the Etherspot Dashboard straight from the BUIDLER component.

## Motivation and Context
-  A Settings Menu item that allows a user to launch the Etherspot Dashboard straight from the BUIDLER component.

## How Has This Been Tested?
- Local Testing
- Jest Test cases

#### Light Theme
image

#### Dark Theme
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/SettingMenu/SettingMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  border-radius: 5px;
+  padding: 12px 16px;
+  font-size: 14px;
+  text-align: left;
+  box-shadow: rgb(0 0 0 / 24%) 0px 3px 8px;
+`;
+
+const MenuItem = styled.div`
+  display: flex;
+  justify-content: left;
```
Comment (MayukhSamadder2000) [inline src/components/SettingMenu/SettingMenu.tsx:29]: `left` I think is depreciated , I think `flex-start` will be a better option

File: src/components/SettingMenu/SettingMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  a,
+  a:visited {
+    color: ${({ theme }) => theme.color.text.topMenu};
+    text-decoration: none;
+  }
+
+  a:hover {
+    color: ${({ theme }) => theme.color.text.settingsIcon};
+    text-decoration: none;
+  }
```
Comment (MayukhSamadder2000) [inline src/components/SettingMenu/SettingMenu.tsx:44]: I think we can create a separate component for the MenuItem anchor something like `MenuItemLink`

File: src/hooks/useOutsideClick.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [ref]);
+};
+
+export default useOutsideAlerter;
```
Comment (MayukhSamadder2000) [inline src/hooks/useOutsideClick.tsx:21]: `useOnClickOutside` seems better over `useOutsideAlerter`

File: src/hooks/useOutsideClick.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react';
+
+const useOutsideAlerter = (ref: any, outsideClickCallback: Function) => {
```
Comment (MayukhSamadder2000) [inline src/hooks/useOutsideClick.tsx:3]: can you try this for ref ? `ref: RefObject<T>`

File: src/components/Menu/MenuItem.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  &:last-child {
+    margin-bottom: 0;
+  }
+`;
+
+const MenuItem = ({ title, icon, onClick }: MenuItemProps) => {
+  return (
+    <>
+      <MenuItemStyled onClick={(event) => onClick(event)}>
+        {icon ? <>{icon}</> : null}
```
Comment (Yash-Chaubal) [inline src/components/Menu/MenuItem.tsx:32]: Probably could just replace this with (also are the empty tags necessary here as there's only one child?)
```suggestion
        {!!icon && <>{icon}</>}
```

File: src/components/Menu/MenuItemAnchor.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  &:last-child {
+    margin-bottom: 0;
+  }
+`;
+
+const MenuItemAnchor = ({ link, title, icon }: MenuItemAnchorProps) => {
+  return (
+    <>
+      <MenuItem>
+        {icon ? <>{icon}</> : null}
```
Comment (Yash-Chaubal) [inline src/components/Menu/MenuItemAnchor.tsx:49]: Same as above

File: src/hooks/useOnClickOutside.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+
+export interface UseRefObj {
+  current: null | HTMLDivElement;
+}
+
+const useOnClickOutside = (ref: UseRefObj, outsideClickCallback: Function) => {
+  useEffect(() => {
+    function handleClickOutside(event: Event) {
```
Comment (Yash-Chaubal) [inline src/hooks/useOnClickOutside.tsx:9]: Could you change this to arrow notation just for consistency? `const a = () => {}`

**Review by Yash-Chaubal — COMMENTED** [review]
Looking good, just added some housework comments

File: src/components/Menu/MenuItem.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  &:last-child {
+    margin-bottom: 0;
+  }
+`;
+
+const MenuItem = ({ title, icon, onClick }: MenuItemProps) => {
+  return (
+    <>
```
Comment (poocart) [inline src/components/Menu/MenuItem.tsx:30]: No need to wrap with fragment `<>`

File: src/components/Menu/MenuItem.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    text-decoration: none;
+  }
+
+  &:last-child {
+    margin-bottom: 0;
+  }
+`;
+
+const MenuItem = ({ title, icon, onClick }: MenuItemProps) => {
+  return (
```
Comment (poocart) [inline src/components/Menu/MenuItem.tsx:29]: can be just component return
```suggestion
const MenuItem = ({ title, icon, onClick }: MenuItemProps) => (
  ...
));
```

File: src/components/Menu/MenuItemAnchor.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  &:last-child {
+    margin-bottom: 0;
+  }
+`;
+
+const MenuItemAnchor = ({ link, title, icon }: MenuItemAnchorProps) => {
+  return (
+    <>
```
Comment (poocart) [inline src/components/Menu/MenuItemAnchor.tsx:47]: both for `<>` and return same as above

File: src/components/Menu/MenuItemAnchor.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+  a:hover {
+    color: ${({ theme }) => theme.color.text.settingsIcon};
+    text-decoration: none;
+  }
+
+  &:hover {
+    color: ${({ theme }) => theme.color.text.settingsIcon};
+    text-decoration: none;
+    #etherspot-logo {
```
Comment (poocart) [inline src/components/Menu/MenuItemAnchor.tsx:32]: can u wrap image as `styled-components` and include into this as `${EtherspotLogp} { someCss }`

Comment (parag-chauhan) [reply src/components/Menu/MenuItemAnchor.tsx:32]: Working on this!

Comment (parag-chauhan) [reply src/components/Menu/MenuItemAnchor.tsx]: @poocart Updated the code!

**Review by Yash-Chaubal — APPROVED** [review]
LGTM 🚀

File: src/components/Menu/MenuItem.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  margin-right: 12px;
+  cursor: pointer;
+
+  :hover {
+    color: ${({ theme }) => theme.color.text.settingsIcon};
+  }
+`;
+
+const MenuItem = ({ title, icon, onClick }: MenuItemProps) => (
+  <MenuItemStyled onClick={(event) => onClick(event)}>
```
Comment (poocart) [inline src/components/Menu/MenuItem.tsx:24]:
```suggestion
  <MenuItemStyled onClick={onClick}>
```

File: src/components/Menu/MenuItemAnchor.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+  margin-bottom: 10px;
+  margin-right: 12px;
+  cursor: pointer;
+  text-decoration: none;
+
+  :visited {
+    color: ${({ theme }) => theme.color.text.topMenu};
```
Comment (poocart) [inline src/components/Menu/MenuItemAnchor.tsx:20]:
```suggestion
  color: ${({ theme }) => theme.color.text.topMenu};

  :visited {
    color: ${({ theme }) => theme.color.text.topMenu};
```

**Review by poocart — APPROVED** [review]
couple suggestions, but lgtm

File: src/hooks/useOnClickOutside.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+
+export interface UseRefObj {
```
Comment (MayukhSamadder2000) [inline src/hooks/useOnClickOutside.tsx:3]: Do we not have any generic type for this?

**Review by MayukhSamadder2000 — APPROVED** [review]
LGTM

**Pull request merged**